programs/bpf_loader/src/lib.rs

```diff
@@ -1490,40 +1490,51 @@ fn execute<'a, 'b: 'a>(
     let log_collector = invoke_context.get_log_collector();
     let transaction_context = &invoke_context.transaction_context;
     let instruction_context = transaction_context.get_current_instruction_context()?;
-    let program_id = *instruction_context.get_last_program_key(transaction_context)?;
+    let (program_id, is_loader_deprecated) = {
+        let program_account =
+            instruction_context.try_borrow_last_program_account(transaction_context)?;
```

Comment:
(note to self: seems there's no risk of bank hash mismatch in this new `RefCell::try_borrow_mut()` outside feature gate.)

Reply:
nod, new code is equivalent to old code, except old code discards some data